### PR TITLE
internal/v5: include content hash header in docker resource downloads

### PR DESCRIPTION
**Pull request by mhilton**

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/charmstore/2226/

File: internal/v5/resources.go [hunk 1]
```diff
@@ -132,7 +134,14 @@ func (h *ReqHandler) serveDownloadResourceDocker(id *router.ResolvedURL, r *mong
 		}
 		resp.Password = password
 	}
-	return httprequest.WriteJSON(w, http.StatusOK, resp)
+	buf, err := json.Marshal(resp)
```
Comment (rogpeppe) [inline internal/v5/resources.go:137]: OK I guess, but please add a comment saying that this is just a temporary hack.

File: internal/v5/resources.go [hunk 2]
```diff
@@ -132,7 +134,14 @@ func (h *ReqHandler) serveDownloadResourceDocker(id *router.ResolvedURL, r *mong
...
+	if err != nil {
+		return errgo.Mask(err)
+	}
+	w.Header().Set("Content-Type", "application/json")
+	w.Header().Set(params.ContentHashHeader, fmt.Sprintf("%x", sha512.Sum384(buf)))
+	_, err = w.Write(buf)
+	return errgo.Mask(err)
```
Comment (rogpeppe) [inline internal/v5/resources.go]: Please don't return an error if the write fails. There's nothing we can do in that case other than produce an inconsistent body.

**Review by rogpeppe — APPROVED** [review]
LGTM I suppose, but I don't really like putting in cruft like this that will stay around forever with no good reason other than the fact it was a bit more convenient at the time.

**Comment by rogpeppe** [comment]
Having looked at how deeply the "resource as bytes" paradigm is embedded in juju, this seems more reasonable to me, so LGTM without reservation this time (other than the error return), thanks.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/charmstore/2240/

**Comment by mhilton** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/charmstore-merge

**Pull request merged**